VMBackup/main/mounts.py

```diff
@@ -33,46 +33,61 @@ def __init__(self, name, type, fstype, mount_point):
         self.type = type
         self.fstype = fstype
         self.mount_point = mount_point
+        self.unique_name = str(self.mount_point) + "_" + str(self.name)
 
 class Mounts:
     def __init__(self,patching,logger):
         self.mounts = []
         added_mount_point_names = [] 
         disk_util = DiskUtil(patching,logger)
         # Get mount points 
-        mount_points, fs_types = disk_util.get_mount_points() 
+        mount_points, mount_points_info = disk_util.get_mount_points() 
         # Get lsblk devices 
         device_items = disk_util.get_device_items(None)
         lsblk_mounts = [] 
-        lsblk_mount_points = [] 
+        lsblk_mount_points = []
+        lsblk_unique_names = []
         lsblk_fs_types = []
         # List to hold mount-points returned from lsblk command but not reurned from mount command 
         lsblk_mounts_not_in_mount = [] 
         for device_item in device_items:
             mount = Mount(device_item.name, device_item.type, device_item.file_system, device_item.mount_point)
             lsblk_mounts.append(mount)
-            logger.log("lsblk mount point "+str(device_item.mount_point)+" added with fs type "+str(device_item.file_system), True)
+            logger.log("lsblk mount point "+str(mount.mount_point)+" added with device-name "+str(mount.name)+" and fs type "+str(mount.fstype)+", unique-name "+str(mount.unique_name), True)
             lsblk_mount_points.append(device_item.mount_point)
+            lsblk_unique_names.append(mount.unique_name)
             lsblk_fs_types.append(device_item.file_system)
             # If lsblk mount is not found in "mount command" mount-list, add it to the lsblk_mounts_not_in_mount array
             if((device_item.mount_point not in mount_points) and (device_item.mount_point not in lsblk_mounts_not_in_mount)):
                 lsblk_mounts_not_in_mount.append(device_item.mount_point)
         # Sort lsblk_mounts_not_in_mount array in ascending order
         lsblk_mounts_not_in_mount.sort()
         # Add the lsblk devices in the same order as they are returned in mount command output
-        for mount_point, fs_type in zip(mount_points, fs_types):
-            if((mount_point in lsblk_mount_points) and (mount_point not in added_mount_point_names)):
-                mountObj = lsblk_mounts[lsblk_mount_points.index(mount_point)]
+        for mount_point_info in mount_points_info:
+            mountPoint = mount_point_info[0]
+            deviceNameParts = mount_point_info[1].split("/")
+            uniqueName = str(mountPoint) + "_" + str(deviceNameParts[len(deviceNameParts)-1])
+            fsType = mount_point_info[2]
+            if((mountPoint in lsblk_mount_points) and (mountPoint not in added_mount_point_names)):
+                lsblk_mounts_index = 0
+                try:
+                    lsblk_mounts_index = lsblk_unique_names.index(uniqueName)
+                except ValueError as e:
```

Comment:
Just for my understanding, mount output being subset of lsblk output this exception shouldn't ideally arise..right ?

Reply:
ideally not, it is a safety check for any such condition where uniquenames have some difference.